### PR TITLE
Compatibility with Swift 4.2

### LINKED ISSUE #150: Compatible with RxSwift 4.3
I found out after update to XCode 10, unable to compile. Due to need to update RxSwift to latest but Action depending on RxSwift 4.2.

### LINKED ISSUE #152: Compatibility with Swift 4.2
Due to new UIKit naming conventions, Action does not compile under Swift 4.2/ XCode 10

### PR DESCRIPTION
**Pull request by stefanomondino**
I've also already updated the Podspec to 3.7 for future cocoapods release

Fixes #150. Fixes #152.

### CODE REVIEW

File: Action.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1051,6 +1057,8 @@
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SWIFT_ACTIVE_COMPILATION_CONDITIONS = DEBUG;
 				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
+				SWIFT_SWIFT3_OBJC_INFERENCE = On;
```
Comment (mAu888) [inline Action.xcodeproj/project.pbxproj]: I think `@objc inference` should be set to default.

File: Action.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1257,6 +1269,8 @@
 				PRODUCT_BUNDLE_IDENTIFIER = com.ashfurrow.Action;
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SKIP_INSTALL = YES;
+				SWIFT_SWIFT3_OBJC_INFERENCE = On;
```
Comment (mAu888) [inline Action.xcodeproj/project.pbxproj]: Same here.

File: Sources/Action/UIKitExtensions/UIAlertAction+Action.swift [hunk 3]
```diff
@@ -5,7 +5,7 @@ import RxCocoa
 
 public extension UIAlertAction {
 
-    public static func Action(_ title: String?, style: UIAlertActionStyle) -> UIAlertAction {
+    public static func Action(_ title: String?, style: UIAlertAction.Style) -> UIAlertAction {
```
Comment (ashfurrow) [inline Sources/Action/UIKitExtensions/UIAlertAction+Action.swift]: Oh shoot, one thing. Is this backwards compatible with older versions of Swift/Xcode/etc? Can we do something like this?

```swift
#if swift(>=4.2)
  typealias ActionStyle = UIAlertAction.Style // or whatever name
#else
  typealias ActionStyle = UIAlertActionStyle
#endif

public static func Action(_ title: String?, style: ActionStyle) -> UIAlertAction {
```

Sorry I didn't catch that earlier – I'd like to stay backwards compatible for a little while 😄

Comment (stefanomondino) [reply Sources/Action/UIKitExtensions/UIAlertAction+Action.swift]: nice catch!

**Comment by ashfurrow** [comment]
Cool, thanks again! I'll get this released to CocoaPods trunk and tag a release on GitHub right away 👍

**Comment by ashfurrow** [comment]
Oh boo, I don't have an adequate Xcode version. Let me download it...

**Comment by ashfurrow** [comment]
Okay all done! Thanks again 👍

**Pull request merged**